From 4fe79dd08c5b0ded32c2120702923573072a3d93 Mon Sep 17 00:00:00 2001
From: Pino Toscano <toscano.pino@tiscali.it>
Date: Mon, 8 May 2023 18:39:36 +0200
Subject: [PATCH] configure.ac: add --enable-examples option

Allow the user to not build the examples through --disable-examples
(enabled by default); this can be useful:
- when cross-compiling, as the examples are not useful
- in distribution builds, not building stuff that is not used in any
  way
---
 Makefile.am  | 5 ++++-
 configure.ac | 5 +++++
 2 files changed, 9 insertions(+), 1 deletion(-)

diff --git a/Makefile.am b/Makefile.am
index deb25268..2e8fede7 100644
--- a/Makefile.am
+++ b/Makefile.am
@@ -4,11 +4,14 @@
 
 AUTOMAKE_OPTIONS = 1.9 gnu dist-xz no-dist-gzip check-news
 
-SUBDIRS = libpam tests libpamc libpam_misc modules po conf examples xtests
+SUBDIRS = libpam tests libpamc libpam_misc modules po conf xtests
 
 if HAVE_DOC
 SUBDIRS += doc
 endif
+if HAVE_EXAMPLES
+SUBDIRS += examples
+endif
 
 CLEANFILES = *~
 
diff --git a/configure.ac b/configure.ac
index b9b0f839..6666b1b2 100644
--- a/configure.ac
+++ b/configure.ac
@@ -224,6 +224,11 @@ AC_ARG_ENABLE([doc],
         WITH_DOC=$enableval, WITH_DOC=yes)
 AM_CONDITIONAL([HAVE_DOC], [test "x$WITH_DOC" = "xyes"])
 
+AC_ARG_ENABLE([examples],
+        AS_HELP_STRING([--disable-examples],[Do not build the examples]),
+        WITH_EXAMPLES=$enableval, WITH_EXAMPLES=yes)
+AM_CONDITIONAL([HAVE_EXAMPLES], [test "x$WITH_EXAMPLES" = "xyes"])
+
 AC_ARG_ENABLE([prelude],
 	AS_HELP_STRING([--disable-prelude],[do not use prelude]),
 	WITH_PRELUDE=$enableval, WITH_PRELUDE=yes)
-- 
2.42.0

